### PR TITLE
Addin some tests to CloudBigtableIOIntegrationTest.java.

### PR DESCRIPTION
**Pull request by sduskis**
This is a portion of the checkins in #672.

This adds additional tests to make sure that the read and write functionality of CloudBigtableIOIntegrationTest works properly.

### CODE REVIEW

File: bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java [hunk 1]
```diff
@@ -74,27 +84,26 @@ public static void shutdown() throws IOException {
     connection.close();
   }
 
+  protected static CloudBigtableTableConfiguration createConfig(String tableId) {
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java]: nit: createTableConfig() ?

File: bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java [hunk 2]
```diff
@@ -102,12 +111,108 @@ public void testWriteToTable_tableExist() throws Exception {
   }
 
   @Test
-  public void testEstimatedAndSplitForSmallTable() throws Exception {
+  public <T> void testWriteToTable_dataWritten() throws Exception {
+    int INSERT_COUNT = 50;
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java]: final int INSERT_COUNT

File: bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java [hunk 3]
```diff
@@ -102,12 +111,108 @@ public void testWriteToTable_tableExist() throws Exception {
...
+      try {
+        writeThroughDataflow(tableName, INSERT_COUNT);
+        checkTableRowCount(tableName, INSERT_COUNT);
+      } finally {
+        admin.deleteTable(tableName);
+      }
+    }
+  }
+
+  protected TableName createNewTable(Admin admin) throws IOException {
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java]: could this be static?

File: bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java [hunk 4]
```diff
@@ -102,12 +111,108 @@ public void testWriteToTable_tableExist() throws Exception {
...
+          Bytes.toBytes(RandomStringUtils.randomAlphanumeric(8)));
+      }
+    });
+    for (int i = 0; i < insertCount; i++) {
+      writer.processElement(mockContext);
+    }
+    writer.finishBundle(null);
+  }
+
+  @SuppressWarnings("unused")
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java]: unused?

File: bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java [hunk 5]
```diff
@@ -102,12 +111,108 @@ public void testWriteToTable_tableExist() throws Exception {
...
+      try {
+        writeViaTable(tableName, INSERT_COUNT);
+        checkTableRowCountViaDataflowResultReader(tableName, INSERT_COUNT);
+      } finally {
+        admin.deleteTable(tableName);
+      }
+    }
+  }
+
+  protected void writeViaTable(TableName tableName, int rowCount) throws IOException {
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java]: Why writeViaTable is protected but checkTableRowCountViaDataflowResultReader is private?
Can we make them all private if they are all indented to be used in this test class?

File: bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java [hunk 6]
```diff
@@ -102,12 +111,108 @@ public void testWriteToTable_tableExist() throws Exception {
   }
 
   @Test
-  public void testEstimatedAndSplitForSmallTable() throws Exception {
+  public <T> void testWriteToTable_dataWritten() throws Exception {
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java]: What is this <T>? remove?

**Comment by sduskis** [comment]
Addressed all of the comments.  PTAL.

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**